src/receive_imf.rs

```diff
@@ -1014,13 +1014,15 @@ async fn add_parts(
             }
         }
 
-        state = if seen
+        state = if (seen && replace_msg_id.is_none())
             || fetching_existing_messages
             || is_mdn
             || is_reaction
             || chat_id_blocked == Blocked::Yes
         {
             MessageState::InSeen
+        } else if seen {
+            MessageState::InNoticed
```

Comment:
The test still passes when I revert this change? Looks like the test doesn't test what it's supposed to test?